services/src/unit-test/java/uk/gov/moj/sdt/services/TargetApplicationSubmissionServiceTest.java

```diff
@@ -1013,106 +951,114 @@ public void processCCDReferenceFailsOnUnSupportedRequest() {
-        EasyMock.verify(mockCacheable);
-        EasyMock.verify(mockErrorMsgCacheable);
-        EasyMock.verify(xmlReader);
-        EasyMock.verify(ccdReferenceValidator);
+        verify(mockConsumerGateway).individualRequest(any(IIndividualRequest.class), anyLong(), anyLong());
     }
 
+    @Test
     public void processCCDReferenceRequestFailOnCaseOffLine() {
         final String sdtRequestRef = "TEST_1";
         final IIndividualRequest individualRequest = new IndividualRequest();
 
-        // Set-up the individual request object
-
         individualRequest.setSdtRequestReference(sdtRequestRef);
         individualRequest.setRequestStatus("Received");
         setUpIndividualRequest(individualRequest);
         individualRequest.setRequestType(JUDGMENT.getType());
         individualRequest.setRequestPayload("Test Xml");
-        expect(xmlReader.getElementValue("Test Xml", "claimNumber")).andReturn("CCD_Reference_number");
-        EasyMock.expectLastCall().times(1);
-        expect(this.ccdReferenceValidator.isValidCCDReference("CCD_Reference_number")).andReturn(true);
-        EasyMock.expectLastCall().times(1);
 
-        expect(this.mockIndividualRequestDao.getRequestBySdtReference(sdtRequestRef)).andReturn(
-            individualRequest);
+        when(this.mockIndividualRequestDao.getRequestBySdtReference(sdtRequestRef)).thenReturn(individualRequest);
 
         final IGlobalParameter individualReqProcessingDelay = new GlobalParameter();
         individualReqProcessingDelay.setValue("10");
         individualReqProcessingDelay.setName("MCOL_INDV_REQ_DELAY");
-        expect(this.mockCacheable.getValue(IGlobalParameter.class, "MCOL_INDV_REQ_DELAY")).andReturn(
+        when(this.mockCacheable.getValue(IGlobalParameter.class, "MCOL_INDV_REQ_DELAY")).thenReturn(
             individualReqProcessingDelay);
 
-        this.mockIndividualRequestDao.persist(individualRequest);
-        EasyMock.expectLastCall();
-
         final IGlobalParameter connectionTimeOutParam = new GlobalParameter();
         connectionTimeOutParam.setName("TARGET_APP_TIMEOUT");
         connectionTimeOutParam.setValue("1000");
-        expect(this.mockCacheable.getValue(IGlobalParameter.class, "TARGET_APP_TIMEOUT")).andReturn(
-            connectionTimeOutParam);
+        when(this.mockCacheable.getValue(IGlobalParameter.class, "TARGET_APP_TIMEOUT")).thenReturn(connectionTimeOutParam);
 
         final IGlobalParameter receiveTimeOutParam = new GlobalParameter();
         receiveTimeOutParam.setName("TARGET_APP_RESP_TIMEOUT");
         receiveTimeOutParam.setValue("12000");
-        expect(this.mockCacheable.getValue(IGlobalParameter.class, "TARGET_APP_RESP_TIMEOUT")).andReturn(
-            receiveTimeOutParam);
+        when(this.mockCacheable.getValue(IGlobalParameter.class, "TARGET_APP_RESP_TIMEOUT")).thenReturn(receiveTimeOutParam);
 
         this.mockCmcConsumerGateway.individualRequest(individualRequest, 1000, 12000);
```

Comment:
This looks as though it needs to be wrapped in a doThrow() to throw a CaseOffLineException.

Reply:
don't need that, not testing the functionality of cmcConsumerGateway, here we need to make sure 
consumerGateway is used instead of mockCmcConsumerGateway when case if OffLine

Reply:
shouldRequestMColGatwayWhenCaseOffLine
processCCDReferenceRequestWhenCaseNotOffLine
Added these two test cases